app/scripts/modules/core/src/pipeline/config/stages/evaluateVariables/EvaluateVariablesExecutionDetails.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+import { StageExecutionLogs, StageFailureMessage } from 'core/pipeline/details';
+import { ExecutionDetailsSection, IExecutionDetailsSectionProps } from 'core/pipeline/config/stages/core';
+
+export interface IEvaluatedVariables {
+  key: string;
+  value: string;
+}
+
+export function EvaluateVariablesExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const {
+    stage: { context = {}, outputs = {} },
+  } = props;
+
+  const evaluatedVariables = context.variables ? (
+    <div>
+      <dl>
+        {context.variables.map((evalPair: IEvaluatedVariables) => (
+          <React.Fragment key={evalPair.key}>
+            <dt>{evalPair.key}</dt>
+            <dd>{outputs[evalPair.key] || '-'}</dd>
+          </React.Fragment>
+        ))}
+      </dl>
+    </div>
+  ) : (
+    <div>
+      <b>No variables were evaluated.</b>
+    </div>
+  );
+
+  return (
+    <ExecutionDetailsSection name={props.name} current={props.current}>
+      {evaluatedVariables}
+      <StageFailureMessage stage={props.stage} message={props.stage.failureMessage} />
+      <StageExecutionLogs stage={props.stage} />
+    </ExecutionDetailsSection>
+  );
+}
+
+export namespace EvaluateVariablesExecutionDetails {
```

Comment:
Not understanding the namespace?

Reply:
I think it just allows the title to be accessible under `EvaluateVariablesExecutionDetails.title`

Reply:
Hmmmmmmmm... I'm not happy with this pattern.   Seems like this should be a static property of the component class.  however, that should happen in a different PR.